Created templates for engage case study and webinar heros

pmahnke opened the pull request:
## Done

- Created templates for heros for engage pages for case studies and webinars

## QA

- Check out this feature branch
- Run the site using the command `./run serve`
- You will need to setup a test page or add one of the below includes on a page in the /engage directory to test these

### Case study headers

#### no image or class

```{% include "engage/shared/_header-case-study.html" with title="Allan Gray unlocks unprecedented availability and productivity with Ubuntu K8s"   %}```

html
```
<section class="p-strip--light">
  <div class="row u-equal-height">
    <div class="col-8">
      <h1>
        Allan Gray unlocks unprecedented availability and productivity with Ubuntu K8s
      </h1>
    </div>
  </div>
</section>
```

![image](https://user-images.githubusercontent.com/441217/49179238-cc447200-f349-11e8-94f3-7bd12afd6099.png)

#### image, no class

```{% include "engage/shared/_header-case-study.html" with title="Allan Gray unlocks unprecedented availability and productivity with Ubuntu K8s"   %}```

html

```
<section class="p-strip--light">
  <div class="row u-equal-height">
    <div class="col-7">
      <h1>
        Allan Gray unlocks unprecedented availability and productivity with Ubuntu K8s
      </h1>
    </div>
    <div class="col-5 u-align--center u-vertically-center">
      <img src="https://assets.ubuntu.com/v1/071489b4-K8s+logo+mobile.svg" alt="image for (( h1 ))">
    </div>
  </div>
</section>
```
![image](https://user-images.githubusercontent.com/441217/49179486-4d036e00-f34a-11e8-84ab-e1025558621a.png)


#### image, class

```{% include "engage/shared/_header-case-study.html" with title="Allan Gray unlocks unprecedented availability and productivity with Ubuntu K8s"  image="071489b4-K8s+logo+mobile.svg" class="test-class" %}```

html
```
<section class="p-strip test-class">
  <div class="row u-equal-height">
    <div class="col-7">
      <h1>
        Allan Gray unlocks unprecedented availability and productivity with Ubuntu K8s
      </h1>
    </div>
    <div class="col-5 u-align--center u-vertically-center">
      <img src="https://assets.ubuntu.com/v1/071489b4-K8s+logo+mobile.svg" alt="image for (( h1 ))">
    </div>
  </div>
</section>
```

![image](https://user-images.githubusercontent.com/441217/49179579-745a3b00-f34a-11e8-9218-eada70061718.png)

### Webinar headers

#### title, subtitle, date, cta, class

```
{% include "engage/shared/_header-webinar.html" with title="What's New in Ubuntu 18.10"  image="d9105909-Dell_XPS_Laptop_Left-Desktop.png?w=413" subtitle="Canonical product managers discuss the new Ubuntu 18.10 release features in this on-demand webinar" date="Webinar is live on 3 December 2018 at 8pm GMT" url="#register-section" cta="Watch the webinar" class="test-class" %}
```

```
<section class="p-strip test-class">
  <div class="row u-equal-height">
    <div class="col-7">
      <h1 style="font-weight: 100;">
        What's New in Ubuntu 18.10
      </h1>
      <h4>
        Canonical product managers discuss the new Ubuntu 18.10 release features in this on-demand webinar
      </h4>
      <p>
        Webinar is live on 3 December 2018 at 8pm GMT
      </p>
      <p>
        <a href="#register-section" class="p-button--neutral">
          Watch the webinar
        </a>
      </p>
    </div>
    <div class="col-5 u-align--center u-vertically-center">
      <img src="https://assets.ubuntu.com/v1/d9105909-Dell_XPS_Laptop_Left-Desktop.png?w=413" alt="image for (( title ))">
    </div>
  </div>
</section>
```
![image](https://user-images.githubusercontent.com/441217/49181028-36f7ac80-f34e-11e8-8429-3b1ca6ce734e.png)


#### no date

```
{% include "engage/shared/_header-webinar.html" with title="What's New in Ubuntu 18.10"  image="d9105909-Dell_XPS_Laptop_Left-Desktop.png?w=413" subtitle="Canonical product managers discuss the new Ubuntu 18.10 release features in this on-demand webinar" url="#register-section" cta="Watch the webinar" %}
```

```
<section class="p-strip--light">
  <div class="row u-equal-height">
    <div class="col-7">
      <h1 style="font-weight: 100;">
        What's New in Ubuntu 18.10
      </h1>
      <h4>
        Canonical product managers discuss the new Ubuntu 18.10 release features in this on-demand webinar
      </h4>
      
      <p>
        <a href="#register-section" class="p-button--neutral">
          Watch the webinar
        </a>
      </p>
    </div>
    <div class="col-5 u-align--center u-vertically-center">
      <img src="https://assets.ubuntu.com/v1/d9105909-Dell_XPS_Laptop_Left-Desktop.png?w=413" alt="image for (( title ))">
    </div>
  </div>
</section>
```
![image](https://user-images.githubusercontent.com/441217/49181071-50005d80-f34e-11e8-9a9c-081e6f2f878f.png)

#### no image, no class

```
{% include "engage/shared/_header-webinar.html" with title="What's New in Ubuntu 18.10" url="#register-section" cta="Watch the webinar"  %}
```

```
{% include "engage/shared/_header-webinar.html" with title="What's New in Ubuntu 18.10"  subtitle="Canonical product managers discuss the new Ubuntu 18.10 release features in this on-demand webinar" date="Webinar is live on 3 December 2018 at 8pm GMT" url="#register-section" cta="Watch the webinar" %}
```

```
<section class="p-strip--light">
  <div class="row u-equal-height">
    <div class="col-8">
      <h1 style="font-weight: 100;">
        What's New in Ubuntu 18.10
      </h1>
      <h4>
        Canonical product managers discuss the new Ubuntu 18.10 release features in this on-demand webinar
      </h4>
      <p>
        Webinar is live on 3 December 2018 at 8pm GMT
      </p>
      <p>
        <a href="#register-section" class="p-button--neutral">
          Watch the webinar
        </a>
      </p>
    </div>
    
  </div>
</section>
```
![image](https://user-images.githubusercontent.com/441217/49181124-71614980-f34e-11e8-8a3b-996811903595.png)

#### no subtitle, no image, no class

```
<section class="p-strip--light">
  <div class="row u-equal-height">
    <div class="col-8">
      <h1 style="font-weight: 100;">
        What's New in Ubuntu 18.10
      </h1>
      
      
      <p>
        <a href="#register-section" class="p-button--neutral">
          Watch the webinar
        </a>
      </p>
    </div>
    
  </div>
</section>
```
![image](https://user-images.githubusercontent.com/441217/49181163-8ccc5480-f34e-11e8-935c-376f5547c545.png)

## Issue / Card

Fixes #4404

webteam-app commented:
Starting demo at: https://www-ubuntu-com-canonical-websites-pr-4450.run.demo.haus/

deadlight commented on templates/engage/shared/_header-case-study.html:
```diff
@@ -0,0 +1,12 @@
+      </h1>
+    </div>
+    {% if image %}<div class="col-5 u-align--center u-vertically-center">
+      <img src="{% if image != "http" %}{{ ASSET_SERVER_URL }}{% endif %}{{ image }}" alt="image for (( title ))" >
```
Nice

anasereijo commented:
Thanks @pmahnke for doing this.
Just some small changes and I think it's ready to go 👍 

#### No image or class

- [ ] because there is just a title and no copy, we can remove the `margin-bottom` from the `h1` by using `class="u-no-margin--bottom"`


#### Image, no class

- [ ] vertically align the title by using `class="row u-equal-height u-vertically-center"`

- [ ] once again, because there is just a title and no copy, we can remove the `margin-bottom` from the `h1` by using `class="u-no-margin--bottom"`

#### Image and class

- [ ] vertically align the title by using `class="row u-equal-height u-vertically-center"`

- [ ] once again, because there is just a title and no copy, we can remove the `margin-bottom` from the `h1` by using `class="u-no-margin--bottom"`

#### Webinar style 
- [ ] vertically align the title by using `class="row u-equal-height u-vertically-center"`

#### Webinar style, no date
- [ ] vertically align the title by using `class="row u-equal-height u-vertically-center"`

#### Webinar style - no image, no class
- [ ] vertically align the title by using `class="row u-equal-height u-vertically-center"`

pmahnke commented:
@anasereijo I think I have done this... seems like

1. class="row u-equal-height u-vertically-center" on all title rows
2. class="u-no-margin--bottom" on all case study h1's

anthonydillon commented on templates/engage/shared/_header-webinar.html:
```diff
@@ -0,0 +1,23 @@
+<section class="{% if class %}p-strip {{ class }}{% else %}p-strip--light{% endif %}">
+  <div class="row u-equal-height u-vertically-center">
```
u-vertically-center should be on the col

anthonydillon commented on templates/engage/shared/_header-webinar.html:
```diff
@@ -0,0 +1,23 @@
+<section class="{% if class %}p-strip {{ class }}{% else %}p-strip--light{% endif %}">
+  <div class="row u-equal-height u-vertically-center">
+    <div class="{% if image %}col-7{% else %}col-8{% endif %}">
+      <h1 style="font-weight: 100;">
```
Maybe a bespoke styling to remove the margin bottom of this is the only-child. Save a lot of null checking logic.

caldav commented:
Small request about the CTA, could we have it on both templates and could we make it so that it only appears on mobile? Because it's mostly used as a fast track to skip the copy and go straight to the form/video. Eg. https://www.ubuntu.com/engage/developer-desktop

pmahnke commented:
case studies

![image](https://user-images.githubusercontent.com/441217/49282459-2565f000-f487-11e8-8481-a2a6a552ff2b.png)

webinars

![image](https://user-images.githubusercontent.com/441217/49282592-75dd4d80-f487-11e8-9d8f-ed79fd3e4ed0.png)

anasereijo commented:
@caldav I really like your idea of always having a CTA on mobile! 👍

deadlight commented:
@pmahnke could you add a file with all of the examples demoed and then rebuild the demo? It's hard for design to look at all of the options.

Something like [this file](https://gist.github.com/deadlight/793f669cd24f5e93ed6275dc48838db0) should work

pmahnke commented:
OK, I have addressed all comments I think... and added a test page [/engage/_test_header](https://www-ubuntu-com-canonical-websites-pr-4450.run.demo.haus/engage/_test_header)

One thing to note, :only-child will not know that a cta is hidden in large/med, so there will be margin-bottom in this case (see the last example on the test page)

caldav commented:
I think it looks great. Just two things:

- There is a lack of spacing between subtitle and CTA when there is no date (which would be typically the case for promoting a webinar after it aired).
- If the CTA is the only difference between both templates, and since it can be declared optionally, maybe we don't need two templates? (cf. https://github.com/canonical-websites/www.ubuntu.com/pull/4450#issuecomment-442984325)

pmahnke commented:
Updated to a single include and added a permanent help page [/engage/_heading-help][https://www-ubuntu-com-canonical-websites-pr-4450.run.demo.haus/engage/_heading-help]
